jax/_src/device_array.py

```diff
@@ -298,6 +303,15 @@ def raise_not_implemented():
 # pylint: enable=protected-access
 
 
+def reconstruct_device_array(fun, args, arr_state, aval_state):
+  """Method to reconstruct a device array from a serialized state."""
+  np_value = fun(*args)
+  np_value.__setstate__(arr_state)
+  jnp_value = jax.device_put(np_value)
```

Comment:
My only slight hesitation here is: which device should we put it on?

But maybe this is fine: you get a device array on the default device, and if you want it somewhere else, put it somewhere else.

This deserves documentation somewhere.

Reply:
I discussed this issue in the PR description

Reply:
I added a note about the behavior to the CHANGELOG. I'm not sure what other obvious place there is to document this, what do you think?

Reply:
I'd suggest either on or near documentation for `DeviceArray`, or as a separate note somewhere.

Reply:
Done, thanks